src/main/scala/org/bblfsh/client/libuast/objtrack.c

```diff
@@ -0,0 +1,60 @@
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "objtrack.h"
+
+#include <jni.h>
+
+#include <stdlib.h>
+
+typedef struct {
+  jobject **vector;
+  size_t used;
+  size_t size;
+} AllocVector;
+
+static AllocVector *_allocVector = NULL;
+
+static void initAllocVector() {
+  const int initialSize = 128;
+
+  _allocVector = (AllocVector *)malloc(sizeof(AllocVector));
+  _allocVector->vector = (jobject **)malloc(initialSize * sizeof(jobject*));
+  _allocVector->used = 0;
+  _allocVector->size = initialSize;
+}
+
+void trackObject(jobject *obj) {
+  if (_allocVector == NULL || _allocVector->vector == NULL) {
+    initAllocVector();
+  }
+
+  if (_allocVector->used == _allocVector->size) {
+    _allocVector->size *= 2;
+    _allocVector->vector = (jobject **)realloc(_allocVector->vector,
+                           _allocVector->size * sizeof(jobject*));
+  }
+  _allocVector->vector[_allocVector->used++] = obj;
+}
+
+void freeObjects() {
+  if (_allocVector == NULL)
+    return; // No object ever tracked thus not initialized
+
+  int i;
```

Comment:
Why is this defined outside the loop context?

Reply:
It was to avoid warnings with Travis' Ubuntu 14.04 GCC about C99 features, but since I added the C99 switch is should be unnecessary, I'll change it.